applications/FluidDynamicsApplication/custom_utilities/cfd_utilities.cpp

```diff
@@ -0,0 +1,429 @@
+        };
+
+    CalculateYPlusAndUTauForConditions(rModelPart, rKinematicViscosityVariable, method);
+
+    KRATOS_CATCH("");
+}
+
+void CalculateYPlusAndUTauForConditionsBasedOnLinearLogarithmicWallFunction(
+    ModelPart& rModelPart,
+    const Variable<double>& rKinematicViscosityVariable,
+    const double VonKarman,
+    const double WallSmoothness,
+    const int MaxIterations,
+    const double Tolerance)
+{
+    KRATOS_TRY
+
+    const std::function<double(array_1d<double, 3>&, const GeometryType&, const array_1d<double, 3>&,
+                               const Vector&, const double, const double, const double)>
+        method = [VonKarman, WallSmoothness, MaxIterations, Tolerance](
+                     array_1d<double, 3>& rFrictionVelocity,
+                     const GeometryType& rGeometry, const array_1d<double, 3>& rNormal,
+                     const Vector& rGaussShapeFunctions, const double Density,
+                     const double KinematicViscosity, const double WallHeight) {
+            const array_1d<double, 3>& r_wall_velocity =
+                EvaluateInPoint(rGeometry, VELOCITY, rGaussShapeFunctions);
+            const array_1d<double, 3>& r_mesh_velocity =
+                EvaluateInPoint(rGeometry, MESH_VELOCITY, rGaussShapeFunctions);
+
+            return CalculateLinearLogarithmicWallFunctionBasedYPlusAndUtau(
+                rFrictionVelocity, r_wall_velocity - r_mesh_velocity, rNormal, KinematicViscosity,
+                WallHeight, VonKarman, WallSmoothness, MaxIterations, Tolerance);
+        };
+
+    CalculateYPlusAndUTauForConditions(rModelPart, rKinematicViscosityVariable, method);
+
+    KRATOS_CATCH("");
+}
+
+template <typename TDataType>
+void DistributeConditionVariableToNodes(ModelPart& rModelPart,
+                                        const Variable<TDataType>& rVariable)
+{
+    KRATOS_TRY
+
+    CalculateNumberOfNeighbourConditions(rModelPart);
+
+    ModelPart::NodesContainerType& r_nodes = rModelPart.Nodes();
+    VariableUtils().SetNonHistoricalVariableToZero(rVariable, r_nodes);
+
+    block_for_each(rModelPart.Conditions(), [rVariable](ConditionType& r_condition) {
+        GeometryType& r_geometry = r_condition.GetGeometry();
+
+        const TDataType& r_value = r_condition.GetValue(rVariable);
+        for (unsigned int i_node = 0; i_node < r_geometry.PointsNumber(); ++i_node)
+        {
+            NodeType& r_node = r_geometry[i_node];
+            const TDataType& r_current_value = r_node.GetValue(rVariable);
+            const double number_of_neighbour_conditions =
+                static_cast<double>(r_node.GetValue(NUMBER_OF_NEIGHBOUR_CONDITIONS));
+
+            r_node.SetLock();
+            r_node.SetValue(rVariable, r_current_value + r_value * (1.0 / number_of_neighbour_conditions));
+            r_node.UnSetLock();
+        }
+    });
+
+    rModelPart.GetCommunicator().AssembleNonHistoricalData(rVariable);
+
+    KRATOS_CATCH("");
+}
```

Comment:
Have you checked that there is no similar util in the core? I guess that @loumalouomega did something similar at some point (possibly I'm wrong...).

If not, I'd add it to the `VariableUtils`.

Reply:
@loumalouomega has implemented extrapolation gauss point values to nodes ryt? or If you have already implemented this, could you point me? Otherwise I will move this to variable utils.